railties/lib/rails/generators/rails/app/templates/config/ci.rb.tt

```diff
@@ -0,0 +1,33 @@
+# Run using bin/ci
+
+CI.run do
+  step "Setup", "bin/setup --skip-server"
+<% unless options.skip_rubocop? %>
+  step "Style: Ruby", "bin/rubocop"
+<% end -%>
+
+  step "Security: Gem audit", "bin/bundler-audit"
+<% if using_node? -%>
+  step "Security: Yarn vulnerability audit", "yarn audit"
+<% end -%>
+<% if options[:javascript] == "importmap" && !options[:api] && !options[:skip_javascript] -%>
+  step "Security: Importmap vulnerability audit", "bin/importmap audit"
+<% end -%>
+<% unless options.skip_brakeman? -%>
+  step "Security: Brakeman code analysis", "bin/brakeman --quiet --no-pager --exit-on-warn --exit-on-error"
+<% end -%>
+<% if options[:api] || options[:skip_system_test] -%>
+  step "Tests: Rails", "bin/rails test"
+<% else %>
+  step "Tests: Rails", "bin/rails test test:system"
```

Comment:
it appears this might be incorrect - running it as is, Rails attempts to test a file named 'test:system' rather than to run both normal tests and system tests, resulting in the following error:

> /.rbenv/versions/3.4.2/lib/ruby/3.4.0/bundled_gems.rb:82:in 'Kernel.require': cannot load such file -- /path/to/rails-project/test:system (LoadError)

Does it make sense to create separate steps for each or to just concat the call? (`rails test && rails test:system`)

Reply:
> cannot load such file -- /path/to/rails-project/test:system (LoadError)

This might be a fallout from https://github.com/rails/rails/pull/54647 (FYI: @Edouard-chin )

Reply:
I also just realized there is a command to run all tests `bin/rails test:all' maybe that's the best option?

Reply:
No, I think test:all mixes all tests together. We want to run all the regular tests first, then the system tests. But we can also just split it into two separate steps?

Reply:
We reverted https://github.com/rails/rails/pull/54647. I think it's fine to leave the `bin/rails test test:system` as is. I'll try to find another solution for #54647.

Sorry for the issue

Reply:
It still might be better to have separate steps if we're thinking about the proposal in https://github.com/rails/rails/pull/54693#issuecomment-2703598874:

> "bin/ci --step rubocop" and "bin/ci --step test" being separate and called from a GH workflow file would be pretty neat

Reason being: folks could use a lighter weight image (with no browsers installed) to run non-system tests.

Reply:
> We reverted https://github.com/rails/rails/pull/54647. I think it's fine to leave the bin/rails test test:system

Actually scratch that. The behaviour of `bin/rails test test:system` just runs the the `test` command and completely ignore `test:system`. Probably best to run it in two steps as DHH mentioned